lock-impl/src/main/java/com/palantir/lock/impl/LockThreadInfoSnapshotManager.java

```diff
@@ -84,7 +72,7 @@ Map<LockDescriptor, LockClientAndThread> getLastKnownThreadInfoSnapshot() {
     public UnsafeArg<Optional<Map<LockDescriptor, LockClientAndThread>>> getRestrictedSnapshotAsOptionalLogArg(
             Set<LockDescriptor> lockDescriptors) {
         String argName = "presumedClientThreadHoldersIfEnabled";
-        if (!threadInfoConfiguration.recordThreadInfo()) {
+        if (!threadInfoConfiguration.current().recordThreadInfo()) {
```

Comment:
We should also clean up the previous value

Reply:
Discussed offline -- this is an interesting usecase for a `BackgroundSupplier`, which would update periodically in the background, and in the case it's disabled, we just return an empty map.

Reply:
However, the given implementation is quite good, so we'll stick with that presently